File: solutions/add-digits/add-digits.py
// @Title: Add Digits
// @Author: realzhangm
// @Date: 2021-10-29T11:10:19+08:00
// @URL: https://leetcode-cn.com/problems/add-digits


class Solution:
    def addDigits(self, num: int) -> int:
        if num < 10:
            return num
        next = 0
        while num > 0:
            next += num%10
            num = num//10
        return self.addDigits(next)

File: solutions/add-digits/add-digits.rs
// @Title: Add Digits
// @Author: realzhangm
// @Date: 2021-10-29T10:58:19+08:00
// @URL: https://leetcode-cn.com/problems/add-digits


impl Solution {
    pub fn add_digits(num: i32) -> i32 {
        (num - 1) % 9 + 1
    }
}
